src/JobConsumer/Program.cs

```
@@ -0,0 +1,316 @@
+                RepoPath = repoPath.Value();
+                DriverPath = driverPath.Value();
+                ServerUrl = serverUrl.Value();
+                ClientUrl = clientUrl.Value();
+
+                var directory = new DirectoryInfo(JobsPath);
+
+                if (!directory.Exists)
+                {
+                    Console.WriteLine($"The path doesn't exist: '{directory.FullName}'");
+                    return -1;
+                }
+
+                if (!File.Exists(DriverPath))
+                {
+                    Console.WriteLine($"The driver could not be found at: '{DriverPath}'");
+                    return -1;
+                }
+
+                // Create the target folders if they don't exist
+                Directory.CreateDirectory(ProcessingPath);
+                Directory.CreateDirectory(ProcessedPath);
+
+                Console.WriteLine("Press enter to exit.");
+
+                while (true)
+                {
+                    // Get oldest file
+                    var nextFile = directory
+                        .GetFiles()
+                        .OrderByDescending(f => f.LastWriteTime)
+                        .FirstOrDefault();
+
+
+                    // If no file was found, wait some time
+                    if (nextFile is null)
+                    {
+                        if (Console.KeyAvailable)
+                        {
+                            if (Console.ReadKey().Key == ConsoleKey.Enter)
+                            {
+                                return 0;
+                            }
+                        }
+
+                        await Task.Delay(1000);
+                        continue;
+                    }
+
+                    Console.WriteLine($"Found '{nextFile.Name}'");
+
+                    // Attempting to move the file to the processing folder in order to lock it
+                    var session = nextFile.Name.Split('.', 2)[0];
```

This code needs to be rock solid, so it should be extremely defensive with lots of logs.

---

The way I see it, it's better to let an exception bubble up and crash the process and let a watchdog observe and report the failure. If there's a bug in the file-moving logic, we cannot reasonably report it back to the PRJobProducer anyway since doing so would rely on the said file-moving logic, having the correct session id, etc...

Not to mention, long term I would love to replace all of this with something like SSH, so I don't want to spend a ton of time over-engineering this.

---

> so I don't want to spend a ton of time over-engineering this.

Like adding an if statement?